data-sourcing-project.md

```diff
@@ -1,76 +1,74 @@
-# Data Dependency Graph in Machine Learning system
+# Data Dependency Graph in Machine Learning systems
 
 ## Introduction
 
 I was told that the team was always open to project proposals.
 
-This is my first time writting such a proposal, and first time writting an academic proposal, it is
-thus probably riddled with issues, please forgive me for that.
+This is my first time writing such a proposal, and the first time that I write an academic proposal at all.
+I beg your pardon if it is (thus) riddled with issues.
 
 ## Problem statement
 
 ["Hidden Technical Debt in Machine Learning Systems
 D Sculley, Gary Holt, Daniel Golovin, Eugene Davydov, Todd Phillips, Dietmar Ebner, Vinay Chaudhary,
 Michael Young, Jean-Francois Crespo, Dan Dennison", NIPS 2015](https://papers.nips.cc/paper/5656-hidden-technical-debt-in-machine-learning-systems.pdf)
-mentions current technical debt in data analysis systems.
+mentions an existing technical debt in data analysis systems.
 
-The hidden data dependency one seemed to me as an interesting and solvable problem.
+The hidden data dependency, in particular, seemed to me as an interesting and solvable problem.
 
-The problem is as follows:
+It is defined as follows:
 
 - Data analysis jobs (whether batch or streaming) are consumers and producers of data.
-- The data are an integral parts of these pipelines, a change in data can have drastic consequences
-  on the correctness of the results of pipelines.
-- Data Dependencies do not have easy solutions to be expressed (to the best of my knowledge), this
+- Data are an integral part of these pipelines: any changes of the input/output data can have drastic consequences
+  on the correctness of the results.
+- Data Dependencies are usually not easily expressed (to the best of my knowledge): this
   can lead to hidden (or even cyclic) dependencies between data sources.
-- Not knowing the dependencies can lead to broken pipelines when schemas are updated (changing a
-  field in a protobuf for example) which may prove costly to fix.
-  This is also an engineering problem as it may make the importance of each data in the whole system
-  troubling to evaluate (which can be used for promoting the data from spindles to SSDs for example).
+- Ignoring such dependencies can lead to broken pipelines when schemas are updated (e.g. changing a
+  field in a protobuf), which may prove to be costly to fix.
+  Plus, this is also an engineering problem as the importance of data in the whole system
+  may appear hard to evaluate (e.g.  promoting the data from spindles to SSDs).
 - The problem of code dependencies is currently solved using static analysis systems.
 
 ## Objective
 
-An objective of this project would be to have a tool that can create a directed graph of all data
+The objective of this project is to develop a tool that can create a directed graph of all data
```

Comment:
"An" here is correct, as the reproducibilty of historical pipelines will also be an objective, even though it is not the main one.

Reply:
I don't like the extreme use of conditional forms ("would", "could", "should", etc). It's like you are not sure at all of what you want to do.